src/Testcontainers/Containers/Modules/Databases/CosmosDbTestcontainer.cs

```diff
@@ -0,0 +1,52 @@
+namespace DotNet.Testcontainers.Containers
+{
+  using System;
+  using System.Net.Http;
+  using System.Threading;
+  using System.Threading.Tasks;
+  using DotNet.Testcontainers.Configurations;
+  using JetBrains.Annotations;
+  using Microsoft.Extensions.Logging;
+
+  [PublicAPI]
+  public sealed class CosmosDbTestcontainer : TestcontainerDatabase
+  {
+    internal CosmosDbTestcontainer(ITestcontainersConfiguration configuration, ILogger logger)
+      : base(configuration, logger)
+    {
+    }
+
+    public HttpMessageHandler HttpMessageHandler => new UriRewriter(this.Hostname, this.Port);
```

Comment:
Do we need this property?

Reply:
This was requested by @JonasBenz so, I guess it has to do with .NET FW and cosmosclient.

Reply:
Ok, but then I would use the property at `HttpClient` too. With the dedicated modules we can move it into a separate class in the future.

Reply:
Done